x/accounts/defaults/lockup/lockup.go

```diff
@@ -356,6 +371,10 @@ func sendMessage(ctx context.Context, msg proto.Message) ([]*codectypes.Any, err
 func (bva *BaseLockup) TrackDelegation(
 	ctx context.Context, balance, lockedCoins, amount sdk.Coins,
 ) error {
+	if amount.IsZero() {
+		return sdkerrors.ErrInvalidCoins.Wrap("delegation attempt with zero coins or insufficient funds")
+	}
+
 	for _, coin := range amount {
```

Comment:
is there anyway to avoid this loop? we should have knowledge of the staking denom or what is being queried?

Reply:
hmmmm yeah i think i can do staking denom query and just need to fetch the amount for that specific denom

Reply:
will do this in the clean up PR

Reply:
i made the change in this [PR](https://github.com/cosmos/cosmos-sdk/pull/20037)